packages/api/cms-api/src/dam/files/file-validation.service.ts

```diff
@@ -37,15 +47,31 @@ export class FileValidationService {
             return `File type and extension mismatch: .${extension} and ${file.mimetype} are incompatible`;
         }
 
-        //svgContainsJavaScript
+        return undefined;
+    }
+
+    async validateFileContents(file: FileUploadInterface): Promise<undefined | string> {
         if (file.mimetype === "image/svg+xml") {
             const fileContent = await readFile(file.path, { encoding: "utf-8" });
 
             if (svgContainsJavaScript(fileContent)) {
+                await this.removeFile(file);
+
                 return "SVG must not contain JavaScript";
             }
         }
 
-        return true;
+        return undefined;
+    }
+
+    // https://github.com/expressjs/multer/blob/master/storage/disk.js#L54-L62
+    private async removeFile(file: FileUploadInterface) {
+        const path = file.path;
+
+        delete (file as Partial<FileUploadInterface>).destination;
+        delete (file as Partial<FileUploadInterface>).filename;
+        delete (file as Partial<FileUploadInterface>).path;
+
+        await unlink(path);
```

Comment:
Don't delete in FileValidationService

Reply:
@nsams FYI: I moved the `removeFile()` method to files.utils.ts and call it in the interceptor (not the FileValidationService) now